katdal/chunkstore_s3.py

```diff
@@ -598,6 +607,19 @@ def complete_request(self, method, url, chunk_name='',
             else:
                 return result
 
+    def _verify_bucket(self, url, chunk_error=None):
+        """Check that bucket associated with `url` exists and is not empty."""
+        bucket = _bucket_url(url)
+        try:
+            response = self.complete_request('GET', bucket, process=lambda r: r)
+        except S3ObjectNotFound as err:
+            # There is no point continuing if the bucket is completely missing
+            raise StoreUnavailable(err) from chunk_error
+        # An empty bucket response has no Contents elements (no need for full XML parsing)
+        if response.ok and b'<Contents>' not in response.content:
+            msg = f'S3 bucket {bucket} is empty - your data is not currently accessible'
```

Comment:
That's what I had originally:
```
S3 bucket <x> is empty. Your data is probably archived to tape - please ask support to restage it.
```
The archive folks did not like it though, since terms like "restage" and "tape" are considered internal implementation details to be abstracted away under ideal circumstances. I added the word "currently" to reduce user panic at the very least :-)